### DISCUSSION TITLE
Height 'auto' unexpected height when splitChannel is defined

### DISCUSSION THREAD

**Discussion by htc36** [katspaugh/wavesurfer.js#3690] [Q&A] [answered]
Hi there,

Love the libary by the way.  I'm using `height: 'auto'` so the waveform fills its parent container (height: 250px). This work fantastically until I want to toggle split channel. Both canvases have `height : auto `and it caluculates height the same way. Where in theory it should do `this.parent.clientHeight / 2 `so it still fits nicely in the parent container.

What its doing when height is set to auto:

https://github.com/katspaugh/wavesurfer.js/assets/54700621/1f465aef-7db2-4a40-b832-70550379e83b

What I want it to do: ( i acheived this by hard coding the heights and divide by 2 if its dual channel)

https://github.com/katspaugh/wavesurfer.js/assets/54700621/7cd12115-c877-4a6c-92de-75ae3a053157

Do you have any suggestions for what i could do? Obvs i could hard code the height instead of using auto. But I'd really rather define the height in a parent component so I don't have to go into wavesurfer config object to adjust it.

I've isolated it down to the getHeight() function

![image](https://github.com/katspaugh/wavesurfer.js/assets/54700621/8ac2d986-505b-4bb8-ae89-77ef8ec6a5de)

Would it be feasable to also take in audioData.numberOfChannels into the calculation. I was thinking a bit of logic: 

```
if (optionsHeight === 'auto') {
 // Insert logic to check overlay is not true on any of splitArray parameter
 if (this.parent.clientHeight && optionsSplitChannel) return this.parent.clientHeight / (audioData.numberOfChannels || 1)

 return this.parent.clientHeight || defaultHeight
}
```

Let me know your thoughts

**Comment by katspaugh** [comment]
Hey, good catch! Dividing by `numberOfChannels` sounds good to me, happy to accept a PR.

**Reply by htc36** [reply, marked as answer]
Created  PR: https://github.com/katspaugh/wavesurfer.js/pull/3705
